### ISSUE TITLE
[BUG] Corrupted generated speeches

### ISSUE THREAD

**Issue by TundraWork** [fishaudio/fish-speech#44]
My fine-tuned model will generate some corrupted speech audios. They all seems like have some words / sentences shuffled.

Here's an example of generated speech:

[fake.zip](https://github.com/fishaudio/fish-speech/files/13817821/fake.zip)

The expected speech text is:

```
这是嘟嘟可，是可莉很久以前就交到的好朋友，要记得他的名字哦！以后别叫他挂在你包上的玩偶了！
```

While the actual speech content in fake.wav is:

```
这是嘟嘟可，<inaudible>记得他的名字哦！以后别<inaudible>就交到的好朋友，要记得他的名字哦！以后别叫他挂在你包上的玩偶了！
```

The inference steps for this output:

```bash
python tools/vqgan/inference.py -i "/mnt/d/Projects/VocalAI/Datasets/GenshinImpact-Voices-Labeled/Klee/vo_JNJEQ003_6_klee_01.wav" --checkpoint-path "results/vqgan_finetune_GenshinImpact_Klee/checkpoints/step_000020000.ckpt"
python tools/llama/generate.py --text "这是嘟嘟可，是可莉很久以前就交到的好朋友，要记得他的名字哦！以后别叫他挂在你包上的玩偶了！" --prompt-text "妈妈说过，今天大家要迎接风神大人，要请他喝酒！如果风神大人高兴，就会变成风来祝福大家！" --prompt-tokens fake.npy --checkpoint-path "results/text2semantic_finetune_spk_GenshinImpact_Klee/checkpoints/step_000001000.ckpt" --config-name text2semantic_finetune_spk_GenshinImpact_Klee --speaker SPK1 --num-samples 2 --compile
python tools/vqgan/inference.py -i codes_0.npy --checkpoint-path "results/vqgan_finetune_GenshinImpact_Klee/checkpoints/step_000020000.ckpt"
```

Given that the VQGAN is relatively stable, and this kind of chaotic output is often seen in an LLM's output, I believe these corrupted speeches are caused by the Llama model.

Is this behavior expected from the model? Do you have any suggestions on how to avoid this kind of issue?

**Comment by leng-yue** [comment]
Can you try generate with`--no-g2p`, seems the generation quality is better without phonemizer.

**Comment by TundraWork** [comment]
The generation quality is slightly better with `--no-g2p`, the shuffling of the sentences was solved, but there are still many inaudible phonemes embedded in the sentence.

An example:

[fake.zip](https://github.com/fishaudio/fish-speech/files/13847702/fake.zip)

**Comment by leng-yue** [comment]
What's the top-5 acc when you are finetuning? Maybe there is some kind of overfitting.

**Comment by TundraWork** [comment]
> What's the top-5 acc when you are finetuning? Maybe there is some kind of overfitting.

![image](https://github.com/fishaudio/fish-speech/assets/8012410/cd9b27c4-9939-408d-9636-48702a0cbdf9)

Yeah I think it's very possible since I used a very small dataset (~100 samples).

**Comment by leng-yue** [comment]
Are u using the latest version + lora? I tuned the dropout ratio to reduce the risk of overfitting.
